src/app/sidebar/sidebar.component.ts

```diff
@@ -38,8 +40,8 @@ export class SidebarComponent implements AfterViewInit {
     }
 
     onLogout() {
-        this.jwtTokenService.logout().pipe(
-            tap(() => this.router.navigate(['/']))
+        this.http.logout().pipe(
+            finalize(() => this.router.navigate(['/']))
```

Comment:
What's the difference between tap and finalize?

Reply:
`tap` is a side-effect operator that doesn't modify the "stream", but returns the same value which it receives [1]. What matters here is that `tap` will only be executed in case of "happy path" – no error result from `logout()` observable.

On the other hand, `finalize` will be executed either on error or non-error value at the end of the observable chain ("stream"). This way, we are sure that this will execute, even in the case of error (_we want that!_).

These two lines were bad code. `logout()` can fail in a few ways [2] and, regardless of the outcome, we want to redirect the user to the landing page.

---
[1] In this case, we don't need the receiver value. That's why there is an empty parameter inside an anonymous function. It could be `tap(logoutResult => console.log(logoutResult))` if we needed to do something with the result. It's usually used to log something or save something in "store", or trigger another action, or do anything as a side-effect.
[2] Missing token when tried to logout user; the user can be already logged out; the user can be offline...